modules/ts/src/ts_gtest.cpp

```diff
@@ -4054,7 +4054,7 @@ enum GTestColor {
   COLOR_YELLOW
 };
 
-#if GTEST_OS_WINDOWS && !GTEST_OS_WINDOWS_MOBILE
+#if GTEST_OS_WINDOWS && !GTEST_OS_WINDOWS_MOBILE && !defined WINRT
```

Comment:
Actually GTest 1.7.0 is the last released version and OpenCV uses it.

OpenCV uses some patches for Google Test to:
- fix build: eliminate warnings about incorrect handling of definitions / make some functions `static`;
- add some functionality (filter by parameter value);
- etc;

I created update for Google Test: https://github.com/Itseez/opencv/pull/4157 . If it helps, we can try to merge it.